front/app/components/Form/Components/Layouts/CLSurveyPageLayout.tsx

```diff
@@ -215,37 +218,55 @@ const CLSurveyPageLayout = memo(
     };
 
     const handleNextAndSubmit = async () => {
-      if (showSubmit && onSubmit) {
-        setIsLoading(true);
-        data.publication_status = 'published';
-        await onSubmit(getFilteredDataForUserPath(userPagePath, data), true);
-        return;
-      }
+      if (!onSubmit) return;
 
       const currentPageCategorization = uiPages[currentStep];
       userPagePath.push(uiPages[currentStep]);
-      if (
-        customAjv.validate(
-          getPageSchema(
-            schema,
-            currentPageCategorization,
-            formState.core?.data,
-            customAjv
-          ),
-          getSanitizedFormData(data)
-        )
-      ) {
-        scrollToTop();
-        data.publication_status = 'draft';
-        data.latest_complete_page = currentStep;
-        onSubmit?.({ data }, false);
-        setCurrentStep(currentStep + 1);
 
-        setIsLoading(false);
-      } else {
+      const isValid = customAjv.validate(
+        getPageSchema(
+          schema,
+          currentPageCategorization,
+          formState.core?.data,
+          customAjv
+        ),
+        getSanitizedFormData(data)
+      );
+
+      if (!isValid) {
         setShowAllErrors?.(true);
         setScrollToError(true);
+        return;
       }
+
+      if (pageVariant === 'after-submission') {
+        const searchParams: Record<string, string> = { show_modal: 'true' };
+        if (phaseId) searchParams.phase_id = phaseId;
+        // if (idea) searchParams.new_idea_id = idea.data.id;
```

Comment:
Didn't figure out yet how to get access to the current idea here. But might not be necessary depending on what we decide wrt the success modal (see slack discussion)